packages/snap/src/dialogs/utxos.tsx

```diff
@@ -0,0 +1,96 @@
+import { constants as libConstants, numberUtils } from '@hathor/wallet-lib';
+import { Bold, Box, Container, Divider, Heading, Section, Text } from '@metamask/snaps-sdk/jsx';
+
+const renderTokenFilterParam = (token) => {
+  if (!token || token === libConstants.NATIVE_TOKEN_UID) {
+    return libConstants.DEFAULT_NATIVE_TOKEN_CONFIG.symbol;
+  }
+
+  return token;
+}
+
+// We show the symbol only if it's HTR in the list
+const renderTokenSymbol = (token) => {
+  if (!token || token === libConstants.NATIVE_TOKEN_UID) {
+    return libConstants.DEFAULT_NATIVE_TOKEN_CONFIG.symbol;
+  }
+
+  return '';
+}
+
+const renderAmountSummary = (data, params) => {
+  if (params.authorities) {
+    let authority = '';
+    if (params.authorities === 1) {
+      authority = 'mint';
+    } else if (params.authorities === 2) {
+      authority = 'melt';
+    }
+
+    return (
+      <Text>{`Authority: ${authority}`}</Text>
+    );
+  }
+
+  return (
+    <Text>{`Amount: ${numberUtils.prettyValue(data.total_amount_available)} ${renderTokenSymbol(params.token)}`}</Text>
+  );
+}
+
+export const utxosPage = async (data, params, origin) => {
+  const content = (
+    <Container backgroundColor='alternative'>
+      <Box>
+        <Heading>UTXO Details</Heading>
+        <Text>
+          {`${origin} requests access to ${data.utxos?.length || 0} UTXOs`}
```

Comment:
suggestion(wording): I think "requests access" may be incorrectly interpreted by users as offering the funds of the UTXO to the origin dApp.

I'd suggest changing this to something like "${origin} requests information about ${n} UTXOs" or "${origin} requests data from ${n} UTXOs"

Reply:
It makes sense, it's better like you suggested. https://github.com/HathorNetwork/hathor-rpc-lib/pull/58/commits/5f5a0f61512dcd7940e2d6bfc0f4c759f9af9015